File: src/main/java/com/thebluehats/server/game/managers/combat/templates/PlayerHitPlayerVerifier.java
package com.thebluehats.server.game.managers.combat.templates;

import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;

public class PlayerHitPlayerVerifier implements EventVerifier<EntityDamageByEntityEvent> {
    @Override
    public boolean verify(EntityDamageByEntityEvent event) {
        return event.getDamager() instanceof Player && event.getEntity() instanceof Player;
    }
}

File: src/main/kotlin/com/thebluehats/server/game/managers/combat/templates/PlayerHitPlayerVerifier.kt
package com.thebluehats.server.game.managers.combat.templates

import org.bukkit.entity.Player
import org.bukkit.event.entity.EntityDamageByEntityEvent

class PlayerHitPlayerVerifier : EventVerifier<EntityDamageByEntityEvent> {
    override fun verify(event: EntityDamageByEntityEvent): Boolean {
        return event.damager is Player && event.entity is Player
    }
}